### PR TITLE
fix(quickstart): Use single tenant (restricted) templates to make deployment easier

### PR DESCRIPTION
**Pull request by jimmidyson**
Hopefully this fixes some of the previous issues with re-deploying.

### CODE REVIEW

**Comment by jimmidyson** [comment]
People! Please review!

File: content/quickstart.md [hunk 1]
```diff
@@ -73,22 +73,29 @@ $ export GITHUB_CLIENT_ID=...
...
+```bash
+$ oc create -f https://raw.githubusercontent.com/syndesisio/syndesis-openshift-templates/master/support/serviceaccount-as-oauthclient-restricted.yml
+```
+
+Deploy syndesis using the following command, replacing "syndesis-restricted-dev" with "syndesis-restricted" depending on the template
 you have just installed:
 
 ```bash
-$ oc new-app syndesis-dev \
+$ oc new-app syndesis-dev-restricted \
```
Comment (aileenc) [inline content/quickstart.md]: restricted-dev not dev-restricted

Comment (jimmidyson) [inline content/quickstart.md]: Thanks! Fixed.

File: content/quickstart.md [hunk 2]
```diff
@@ -73,22 +73,29 @@ $ export GITHUB_CLIENT_ID=...
...
     -p ROUTE_HOSTNAME=syndesis.$(minishift ip).xip.io \
     -p OPENSHIFT_MASTER=$(oc whoami --show-server) \
+    -p OPENSHIFT_PROJECT=$(oc project -q) \
+    -p OPENSHIFT_OAUTH_CLIENT_SECRET=$(oc sa get-token syndesis-oauth-client) \
     -p GITHUB_OAUTH_CLIENT_ID=${GITHUB_CLIENT_ID} \
-    -p GITHUB_OAUTH_CLIENT_SECRET=${GITHUB_CLIENT_SECRET} \
-    -p INSECURE_SKIP_VERIFY=true
+    -p GITHUB_OAUTH_CLIENT_SECRET=${GITHUB_CLIENT_SECRET}
 ```
```
Comment (dsimansk) [inline content/quickstart.md]: Are cert fixed on Minishift now? Will it work without `INSECURE_SKIP_VERIFY=true`?

Comment (jimmidyson) [inline content/quickstart.md]: It worked for me... Would appreciate it if you could try it out and see please.

Comment (dsimansk) [inline content/quickstart.md]: Sure, I let you know in a few minutes.

File: content/quickstart.md [hunk 3]
```diff
@@ -73,22 +73,29 @@ $ export GITHUB_CLIENT_ID=...
 $ export GITHUB_CLIENT_SECRET=...
 ```
 
-Install the OpenShift template (syndesis-dev.yml or syndesis.yml as discussed [above]({{< relref "#template-selection" >}})):
+Install the OpenShift template (syndesis-restricted-dev.yml or syndesis-restricted.yml as discussed [above]({{< relref "#template-selection" >}})):
 
 ```bash
-$ oc create -f https://raw.githubusercontent.com/syndesisio/syndesis-openshift-templates/master/syndesis-dev.yml
+$ oc create -f https://raw.githubusercontent.com/syndesisio/syndesis-openshift-templates/master/syndesis-restricted-dev.yml
 ```
```
Comment (dsimansk) [inline content/quickstart.md]: The raw URL should be:
https://raw.githubusercontent.com/syndesisio/syndesis-openshift-templates/master/syndesis-dev-restricted.yml

File: content/quickstart.md [hunk 4]
```diff
@@ -73,22 +73,29 @@ $ export GITHUB_CLIENT_ID=...
...
+```bash
+$ oc create -f https://raw.githubusercontent.com/syndesisio/syndesis-openshift-templates/master/support/serviceaccount-as-oauthclient-restricted.yml
+```
+
+Deploy syndesis using the following command, replacing "syndesis-restricted-dev" with "syndesis-restricted" depending on the template
 you have just installed:
 
 ```bash
-$ oc new-app syndesis-dev \
+$ oc new-app syndesis-restricted-dev \
```
Comment (dsimansk) [inline content/quickstart.md]: Template name states:
```yaml
metadata:
  name: syndesis-dev-restricted
```

File: content/quickstart.md [hunk 5]
```diff
@@ -73,22 +73,29 @@ $ export GITHUB_CLIENT_ID=...
 $ export GITHUB_CLIENT_SECRET=...
 ```
 
-Install the OpenShift template (syndesis-dev.yml or syndesis.yml as discussed [above]({{< relref "#template-selection" >}})):
+Install the OpenShift template (syndesis-restricted-dev.yml or syndesis-restricted.yml as discussed [above]({{< relref "#template-selection" >}})):
 
 ```bash
-$ oc create -f https://raw.githubusercontent.com/syndesisio/syndesis-openshift-templates/master/syndesis-dev.yml
+$ oc create -f https://raw.githubusercontent.com/syndesisio/syndesis-openshift-templates/master/syndesis-restricted-dev.yml
```
Comment (rhuss) [inline content/quickstart.md]: I wonder whether we should always point to `master` or to a released version of the templates (where we also have pinned the image versions) ?

File: content/quickstart.md [hunk 6]
```diff
@@ -98,5 +105,3 @@ $ watch oc get pods
 ```
 
 You should now be able to open `https://syndesis.$(minishift ip).xip.io` in your browser.
```
Comment (rhuss) [inline content/quickstart.md]: Any idea when a switch to `.nip` is possible ? Wonder whether there is already an issue open at minishift wrt to the problematic certs ? (still not really clear what the issue is and whether it part of `oc cluster up` or `minishift)

File: content/quickstart.md (hunk 4 above)
Comment (rhuss) [inline content/quickstart.md]: It's `syndesis(-dev)?(-restricted)?(-ephemeral)?.yml`, in that order.

File: content/quickstart.md (hunk 2 above)
Comment (dsimansk) [inline content/quickstart.md]: Yes it worked for me too. 👍

File: content/quickstart.md (hunk 6 above)
Comment (dsimansk) [inline content/quickstart.md]: I recall that [actual PR](https://github.com/openshift/origin/commit/d7d3b7c67dbeea844389bf2d23aa8e4ded6088cc) was meant to go to 3.6 release.

File: content/quickstart.md (hunk 5 above)
Comment (jimmidyson) [inline content/quickstart.md]: We should cut a release soon but for now master is fine - we have no release to point at yet! But as soon as we do...

**Comment by dsimansk** [comment]
@jimmidyson I have one more question. Do we expect that user is using `oc login -u developer:developer` account or `oc login -u system:admin`? Because my muscle memory goes for `system:admin` all the time. Should we mention it explicitly?

**Comment by jimmidyson** [comment]
@dsimansk developer should be enough, that's why I switched to restricted templates so we don't require admin karma. Doesn't minishift automatically log the user in as developer? Can't remember. Either user will work of course.

**Comment by dsimansk** [comment]
@jimmidyson ehm, yes there's auto login actually. Developer user worked for me also.

**Comment by jimmidyson** [comment]
@dsimansk Cool thanks!

**Pull request merged**